src/utils.py

```diff
@@ -86,8 +88,8 @@ class ServerHTTPSConnection(http.client.HTTPSConnection):
     def __init__(self, config, server, port, cert_file):
         self.no_ssl = config.suppress_ssl or not FEAT_SSL
         if self.no_ssl:
-            if config.use_proxy == True:
-                http.client.HTTPSConnection.__init__(self, config.proxy_url, config.proxy_port, context=context)
```

Comment:
By removing the assignment of the context, in this constructor, does it compromise the security of the HTTPS connection in any way?

See:

- https://docs.python.org/3.4/library/http.client.html#http.client.HTTPSConnection
- https://docs.python.org/3.4/library/ssl.html#ssl-security